ush/gfs_bufr.sh

```diff
@@ -45,30 +54,37 @@ cat << EOF > gfsparm
  &NAMMET
   levs=${LEVS},makebufr=${bufrflag},
   dird="${COM_ATMOS_BUFR}/bufr",
-  nstart=${FSTART},nend=${FEND},nint=${FINT},
+  nstart=${fhr},nend=${fhr},nint=${FINT},
   nend1=${NEND1},nint1=${NINT1},nint3=${NINT3},
-  nsfc=80,f00=${f00flag},fformat=${fformat},np1=0
+  nsfc=80,f00=${f00flag},fformat=${fformat},np1=0,
+  fnsig="sigf${fhr}",
+  fngrib="flxf${fhr}", 
+  fngrib2="flxf${fhr_p}" 
 /
 EOF
 
 sleep_interval=10
 max_tries=1000
-for (( hr = 10#${FSTART}; hr <= 10#${FEND}; hr = hr + 10#${FINT} )); do
-   hh2=$(printf %02i "${hr}")
-   hh3=$(printf %03i "${hr}")
-
-   #---------------------------------------------------------
-   # Make sure all files are available:
-   filename="${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atm.logf${hh3}.${logfm}"
-   if ! wait_for_file "${filename}" "${sleep_interval}" "${max_tries}"; then
-     echo "FATAL ERROR: COULD NOT LOCATE logf${hh3} file"
-     exit 2
-   fi
-   
-   #------------------------------------------------------------------
-   ${NLN} "${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atmf${hh3}.${atmfm}" "sigf${hh2}"
-   ${NLN} "${COM_ATMOS_HISTORY}/${RUN}.${cycle}.sfcf${hh3}.${atmfm}" "flxf${hh2}"
-done
+
+#---------------------------------------------------------
+# Make sure all files are available:
+
+filename="${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atm.logf${fhr}.${logfm}"
+if ! wait_for_file "${filename}" "${sleep_interval}" "${max_tries}"; then
+  echo "FATAL ERROR: COULD NOT LOCATE logf${fhr} file"
+  exit 2
+fi
+
+filename="${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atm.logf${fhr_p}.${logfm}"
+if ! wait_for_file "${filename}" "${sleep_interval}" "${max_tries}"; then
+  echo "FATAL ERROR: COULD NOT LOCATE logf${fhr_p} file"
+  exit 2
```

Comment:
Same as above. I think we can just check if the files exist without using `wait_for_file()`.

Reply:
Fixed it.